mimesis/providers/person.py

```diff
@@ -253,11 +259,24 @@ def email(self, domains: Union[tuple, list] = None) -> str:
 
         domain = self.random.choice(domains)
         name = self.username(template='ld')
-        return '{name}{domain}'.format(
+        email = '{name}{domain}'.format(
             name=name,
             domain=domain,
         )
 
+        if unique:
+            while email in self._previously_generated['emails']:
+                domain = self.random.choice(domains)
+                name = self.username(template='ld')
+                email = '{name}{domain}'.format(
+                    name=name,
+                    domain=domain,
+                )
+
+        self._previously_generated['emails'].add(email)
```

Comment:
As I say above, I think it will better to use `self._store` here, like this: `self._stor['emails'].add(email)`, but if you do not agree, we can discuss it.